api/agent/call.go

```diff
@@ -205,11 +206,24 @@ func FromModel(mCall *models.Call) CallOpt {
 	return func(a *agent, c *call) error {
 		c.Call = mCall
 
-		// NOTE this adds content length based on payload length
 		req, err := http.NewRequest(c.Method, c.URL, strings.NewReader(c.Payload))
 		if err != nil {
 			return err
 		}
+
+		// Hoist original request content type into async request
+		if req.Header.Get("Content-Type") == "" {
+			content_type, ok := c.EnvVars["Fn_header_content_type"]
+			if ok {
+				req.Header.Set("Content-Type", content_type)
+			}
+		}
+
+		// Ensure content-length in async requests for protocol/http DumpRequestTo()
+		if req.Header.Get("Content-Length") == "" {
```

Comment:
`req.ContentLength` suites better

Reply:
since async is the only one using this (and tests) i think this will end up working but it seems brittle still to rewrite blank content length to a value (len(c.Payload)) that is always 0 for sync requests, this would break them most likely (client will think it's an empty body). 

am ok with just patching content length in wait of a more proper header fix, we really need to fix the headers (they shouldn't have `Fn_header_` for routes with "http" format)